python/src/scipp/plotting/model.py

```diff
@@ -198,6 +202,57 @@ def _string_tick_formatter(self, array_values, size):
         return lambda val, pos: array_values[int(val)] if (int(
             val) >= 0 and int(val) < size) else ""
 
+    def _date_tick_formatter(self, offset, dim):
+        """
+        Format datetime ticks: adjust the time precision and update
+        offset according to the currently displayed range.
+        """
+        def formatter(val, pos):
+            dt = str(np.datetime64(int(val) + int(offset), 'ns'))
+            start = 0
+            end = len(dt)
+            u = ""
+            suffix = ""
+            bounds = self.interface["get_view_axis_bounds"](dim)
+            diff = bounds[1] - bounds[0]
+            if diff < 2e3:  # offset: 2017-01-13T12:15:45.123456, tick: 789 ns
+                start = 26
+                u = "ns"
+            elif diff < 2e6:  # offset: 2017-01-13T12:15:45.123, tick: 456 us
+                start = 23
+                end = 26
+                u = "us"
+            elif diff < 2e9:  # offset: 2017-01-13T12:15, tick: 45.123 s
```

Comment:
No `ms` between `s` and `us`?

Reply:
No.
I did have it at one point but decided it was clearer for the user to see `45.123 s` instead of offset `2017-01-13T12:15:45` and a tick: `123 ms`
But this is of course open for discussion.
Just try it out and see what feels best

Reply:
Yes, makes sense. Maybe add a comment that this is intentional, to ensure it does not get "fixed" later?